### PR TITLE
Added support for https

### PR DESCRIPTION
**Pull request by angelnu**
Added support for https streams

### CODE REVIEW

**Comment by angelnu** [comment]
I made a mistake and used https in both cases: fixing and updating pull-request

**Comment by angelnu** [comment]
I also added tests for https

File: test/client.js [hunk 1]
```diff
@@ -64,4 +65,62 @@ describe('Client', function () {
...
     });
   });
+  describe('fixtures: icy-server-https-response', function () {
+      it('should not get a Parse Error with https server', function (done) {
+        var options = {
+            key: fs.readFileSync('test/fixtures/key.pem'),
+            cert: fs.readFileSync('test/fixtures/cert.pem')
+        };
+        //Self signed certificate
+        process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
```
Comment (TooTallNate) [inline test/client.js]: Change this to `rejectUnauthorized: false` in the `options` object. No need to touch the process env.

**Comment by angelnu** [comment]
I change the test to rejectUnauthorized: false for the first test case. In the second one, my objective is to test the Client with an string as input so I have to change the global setting there.

**Comment by angelnu** [comment]
The reason for the second test is to actually test the string part of the code:
```
function Client (options, cb) {
  var req;
  if ((typeof options == 'string' && options.indexOf("https")==0) ||
      (typeof options == 'object' && options.protocol == "https:"))
```

Another option would be to do:
```
function Client (options, cb) {
  if (typeof options == 'string')
      options = url.parse(options)
   var req;
   if (options.protocol == "https:")
```
What is your preference?

**Comment by TooTallNate** [comment]
I'd prefer the `url.parse()` call to be in the Client constructor as well, which would more or less render the "string" test unnecessary.

**Comment by angelnu** [comment]
Right. I changed the client code and removed the second testcase.

**Comment by TooTallNate** [comment]
Sweet. Looks good to me. Thanks! I guess I never ran into an https icecast stream before. Is it a public one out of curiosity?

**Comment by angelnu** [comment]
Thanks for merging!

It is a private stream served by my home apache/icecast setup. This way I only have to open/secure one single port.

**Comment by TooTallNate** [comment]
v2.1.0 published to npm

**Comment by angelnu** [comment]
Thanks!
Picked up the new version in my project at https://github.com/angelnu/ioBroker.chromecast

**Pull request merged**